docs/pages/ds-audio_L3_Python_TestSpecification.md

```diff
@@ -0,0 +1,143 @@
+# Audio Settings L3 Python Test Specification
+
+## Table of Contents
+
+- [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+- [Overview](#overview)
+- [References](#references)
+- [Level 3 Python Tests](#level-3-python-test-cases-high-level-overview)
+
+### Acronyms, Terms and Abbreviations
+
+- `HAL`    - Hardware Abstraction Layer
+- `API`    - Application Programming Interface
+- `L2`     - Level 2 Testing
+- `L3`     - Level 3 Testing
+- `DS`     - Device Settings
+- `DUT`    - Device Under Test
+- `NA`     - Not Applicable
+- `RAFT`   - Rapid Automation Framework for Testing
+
+## Overview
+
+This document describes the L3 Test Procedure for the Device Settings Audio module.
+
+## References
+
+- dsAudio HAL Interface - [dsAudio.h](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsAudio.h)
+- High Level Test Specification - [ds-audio_High-Level_TestSpecification.md](https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/docs/pages/ds-audio_High-Level_TestSpecification.md)
+- `RAFT` - [`RAFT`](https://github.com/rdkcentral/python_raft/)
+- ut_raft - [ut_raft](https://github.com/rdkcentral/ut-raft)
+
+## Level 3 Python Test Cases High Level Overview
+
+The class diagram below illustrates the flow of dsAudio L3 Python test cases:
+
+```mermaid
+---
+title: dsAudio - Python Class Flow
```

Comment:
**YAML File Inputs**

* **config.yaml**
    * Identifies the rack configuration and platform used
    * References platform-specific config from `deviceConfig.yaml`

* **deviceConfig.yaml**
    * Specifies overall configuration for the platform
    * Can be overridden by:
        * Changing locally `.yaml` file directory
        * Using `--deviceConfig` command line switch

* **componentProfile.yaml/platformProfile.yaml** (Implied)
    * Contains component-specific configurations
    * Contains platform wide configuration broken down into seperate components

**Test Execution**

1. User runs test
    * Chooses platform via `--config config.yaml` and config file
    * `--slot 1` is default (optional)

2. Test extracts:
    * Specific component configuration
    * `validationProfile` for the platform

3. Extracted configurations drive `ComponentClass` capabilities

Reply:
Something like this, we need to connect the input to the config also

```mermaid
classDiagram
    testControl <|-- ut_raft
    class ut_raft{
        class~UTMenuNavigator~ menuParsering
        class~utUserResponse~ useResponse
        class~utPlayer~ player
        class~utConfigReader~ configReader
        - config: Config  
        - deviceConfig: DeviceConfig
        - componentProfile: ComponentProfile
        - platformProfile: PlatformProfile
    }
    ut_raft <|-- dsAudio
    dsAudio <|-- L3TestClasses

    class Config {
        - rackConfiguration
        - platform
    }

    class DeviceConfig {
        - cpePlatformConfiguration
    }

    class PlatformProfile {
        - componentConfigurations
    }
    PlatformProfile <|-- DeviceConfig
    DeviceConfig <|-- Config
```

Reply:
I think you should just state that ut_raft is in the way, and remove UTMenuNavigator etc from the class.

Because it changes all the time and will be instantly out of date otherwise.

Reply:
done